app/src/main/java/tech/relaycorp/gateway/domain/LocalConfig.kt

```diff
@@ -25,8 +27,9 @@ class LocalConfig
         sensitiveStore.read(PRIVATE_KEY_FILE_NAME)
             ?.toPrivateKey()
             ?.toKeyPair()
-            ?: generateKeyPair()
-                .also { setKeyPair(it) }
+            ?: throw RuntimeException("No key pair was found")
```

Comment:
The issue I described earlier, where a second key pair was generated and saved, made me realise that we shouldn't be gracefully doing this. This key pair should only be generated once in the lifetime of the app.

Having said this, I wasn't sure how to fail when this method was called before the key pair was available, so a `RuntimeException` might not be a good option.

Reply:
RuntimeException is fine, it's pretty clear it's not something we want to handle.